### DISCUSSION TITLE
WhiteList only

### DISCUSSION THREAD

**Discussion by OneGoogus** [lloesche/valheim-server-docker#125] [Q&A]
How do you configurate the server as a no password and whitelist only server?

**Comment by lloesche** [comment]
I have never tried it, but I suppose you could see what happens if you start the server with:
```
VALHEIM_PLUS=true
SERVER_PASS=""
SERVER_PUBLIC=false
VPCFG_Server_enabled=true
VPCFG_Server_enforceMod=false
VPCFG_Server_disableServerPassword=true
```
and add the client IDs to your `/config/permittedlist.txt` file.

**Comment by kachunkachunk** [comment]
I currently run with this, and just wanted to confirm that it works swimmingly.

Small note:
If you have `public -1` set among server runtime arguments, it will eventually lead to users trying to connect to what shows up as a password-less Valheim server. It's not really an issue, unless a stranger discovers an exploit or bug to bypss the permitted list and get in.
I currently ship and inspect server logs to trend what versions connect, in the off chance that I might notice a new version appear. That'd prompt me that a server update may be ready.

Otherwise:
- Users _not_ in the permittedlist.txt file (randoms, friends you haven't added to your list) may see a "Banned" notification upon connecting. That can be alarming for some folks. ;)
- If Valheim Plus is enforcing the mod with `enforceMod=true`, and/or the connecting client is using a different version, they first see an `Incompatible Version` error upon connecting, and won't see a "Banned" message.

You can look up 17-digit Steam IDs [here](https://steamidfinder.com) or [here](https://steamid.io/lookup)

Edit: I'd suggest following a scheme like:
```
// List permitted players ID ONE per line
// Friend A
76561234567890123
// Friend B
76561234567890124
```

**Reply by lloesche** [reply]
Thank you for confirming @kachunkachunk !
Preserving this as a discussion for others to find.

Also added the `SERVER_PUBLIC=false` suggestion.
